src/storage/table/variant_column_data.cpp

```diff
@@ -278,10 +278,11 @@ void VariantColumnData::FetchRow(TransactionData transaction, ColumnFetchState &
 		auto intermediate = CreateUnshreddingIntermediate(result_idx + 1);
 		auto &child_vectors = StructVector::GetEntries(intermediate);
 		// fetch the validity state
-		validity->FetchRow(transaction, *state.child_states[0], row_id, result, result_idx);
+		validity->FetchRow(transaction, *state.child_states[0], storage_index, row_id, result, result_idx);
```

Comment:
It *looks* like `child_states` in `ColumnFetchState` is not needed at all, all fetches at all levels can just use the base fetch state